src/utils/store-utils.ts

```diff
@@ -210,6 +210,46 @@ export function parseTsvContent(content: string): { headers: string[]; data: str
   return { headers, data };
 }
 
+/**
+ * Applies data type changes to a column draft (for use with Immer produce).
+ * It mutates the draft directly.
+ */
```

Comment:
Thanks for the comment (missed this on first read), but I don't think that's the only way to make a util function work with immer. We should definitely find a way to not introduce util functions with side effects here - also makes for pretty awkward tests. I left a comment on the store hook that calls this on why I think we can just return the update here as a pure function and then handle the assembly in the store.